application-admintools-default/src/main/java/com/xwiki/admintools/internal/files/resources/LogsDataResource.java

```diff
@@ -0,0 +1,201 @@
+import com.xwiki.admintools.download.DataResource;
+import com.xwiki.admintools.internal.data.identifiers.CurrentServer;
+
+import static java.lang.Integer.parseInt;
+
+/**
+ * {@link DataResource} implementation for accessing log files.
+ *
+ * @version $Id$
+ * @since 1.0
+ */
+@Component
+@Named(LogsDataResource.HINT)
+@Singleton
+public class LogsDataResource implements DataResource
+{
+    /**
+     * Component identifier.
+     */
+    public static final String HINT = "logsDataResource";
+
+    private static final String FROM_DATE_FILTER_KEY = "from";
+
+    private static final String TO_DATE_FILTER_KEY = "to";
+
+    private static final String ERROR_SOURCE = " Root cause is: [{}]";
+
+    @Inject
+    private Logger logger;
+
+    @Inject
+    private CurrentServer currentServer;
+
+    @Inject
+    private Provider<XWikiContext> contextProvider;
+
+    @Override
+    public String getIdentifier()
+    {
+        return HINT;
+    }
+
+    @Override
+    public byte[] getByteData(String input) throws Exception
+    {
+        try {
+            File file = new File(currentServer.getCurrentServer().getLastLogFilePath());
+
+            try (RandomAccessFile randomAccessFile = new RandomAccessFile(file, "r")) {
+                int lines = parseInt(input);
+
+                long fileLength = randomAccessFile.length();
+                List<String> logLines = new ArrayList<>();
+
+                // Calculate the approximate position to start reading from based on line length.
+                long startPosition = fileLength - 1;
+                for (long i = 0; i < lines && startPosition > 0 && i < 50000; startPosition--) {
+                    randomAccessFile.seek(startPosition - 1);
+                    int currentByte = randomAccessFile.read();
+                    if (currentByte == '\n' || currentByte == '\r') {
+                        // Found a newline character, add the line to the list.
+                        logLines.add(randomAccessFile.readLine());
+                        i++;
+                    }
+                }
+                // Reverse the list to get the lines in the correct order.
+                Collections.reverse(logLines);
+                // Join the lines with newline characters.
+                return String.join("\n", logLines).getBytes();
+            }
+        } catch (IOException exception) {
+            String errMessage =
+                String.format("Could not find log files at %s.", currentServer.getCurrentServer().getLastLogFilePath());
+            logger.warn(errMessage + ERROR_SOURCE, ExceptionUtils.getRootCauseMessage(exception));
+            throw new IOException(errMessage, exception);
+        } catch (RuntimeException exception) {
+            String errMessage = "Failed to retrieve logs.";
+            logger.warn(errMessage + ERROR_SOURCE, ExceptionUtils.getRootCauseMessage(exception));
+            throw new RuntimeException(errMessage, exception);
```

Comment:
Here you are sending the error message along with the exception, so you don't need to log it in here. The method that really handles this exception will need to get the root cause and log it
Also, RuntimeException is still too generic, it does not give enough information on what went wrong. Both the message and the exception need to be more specific